lib/themes/dosomething/paraneue_dosomething/js/campaign/Graph.js

```diff
@@ -1,39 +1,150 @@
-import $ from 'jquery';
 import Chart from 'chart.js';
 
 var lineGraph = function($el, goal, data) {
+
   // Extend the line graph to create a new chart type that
-  // draws a goal line across the chart.
+  // draws a goal line across the chart and highlights certain points
+  // in the dataset with a point dot.
   Chart.types.Line.extend({
     name: "LineWithGoal",
-    initialize: function () {
-      Chart.types.Line.prototype.initialize.apply(this, arguments);
+
+    initialize: function (data) {
+      // Chart.types.Line.prototype.initialize.apply(this, arguments);
+      var helpers = Chart.helpers;
+      this.datasets = [];
+
+      // Extend the default point class, so we can pass options to the constructor,
+      // For example, if this particular point show be displayed with a dot or not
+      this.PointClass = Chart.Point.extend({
+        strokeWidth: this.options.pointDotStrokeWidth,
+        radius: this.options.pointDotRadius,
+        display: false,
+        hitDetectionRadius: this.options.pointHitDetectionRadius,
+        ctx: this.chart.ctx,
+        inRange: function(mouseX) {
+          return (Math.pow(mouseX - this.x, 2) < Math.pow(this.radius + this.hitDetectionRadius, 2));
+        }
+      });
+
+      // Iterate through each of the datasets.
+      helpers.each(data.datasets, function(dataset) {
+        // Empty dataset where we will put the new points we create.
+        var datasetObject = {
+          label: dataset.label || null,
+          fillColor: dataset.fillColor,
+          strokeColor: dataset.strokeColor,
+          pointColor: dataset.pointColor,
+          pointStrokeColor: dataset.pointStrokeColor,
+          points: []
+        };
+        // Build this into a property of the chart.
+        this.datasets.push(datasetObject);
+
+
+        // Iterate through each of the labels and create a data point for each one.
+        // For each point, specify if we want it's label to display or not.
+        helpers.each(data.labels, function(label, index) {
+          var displayPointDot;
+          //@TODO - Programmatically find these points and clean this up.
+          if (label === "JAN 1" || label === "JAN 10" || label === "JAN 21" || label === "JAN 31") {
```

Comment:
Ah yea, I see what you are getting at. I can look into that when we I start getting actual data from the backend to see what makes sense.